src/measure_theory/integral/lebesgue.lean

```diff
@@ -2240,6 +2240,110 @@ begin
   exact funext (λ a, lintegral_dirac a f),
 end
 
+-- Q: Is this instance a bad idea? It is certainly always the correct one, but does it
+-- affect performance?
+@[priority 100]
+instance measurable_space.top.measurable_singleton_class {α : Type*} :
+  @measurable_singleton_class α (⊤ : measurable_space α) :=
+{ measurable_set_singleton := λ i, measurable_space.measurable_set_top, }
+
+lemma _root_.ennreal.tsum_const_eq [measurable_singleton_class α] (c : ℝ≥0∞) :
+  (∑' (i : α), c) = (measure.count (univ : set α)) * c :=
+by rw [← lintegral_count, lintegral_const, mul_comm]
+
+/-- Markov's inequality for the counting measure with hypothesis using `tsum` in `ℝ≥0∞`. -/
+lemma _root_.ennreal.count_const_le_le_of_tsum_le [measurable_singleton_class α]
+  {a : α → ℝ≥0∞} (a_mble : measurable a) {c : ℝ≥0∞} (tsum_le_c : ∑' i, a i ≤ c)
+  {ε : ℝ≥0∞} (ε_ne_zero : ε ≠ 0) (ε_ne_top : ε ≠ ∞) :
+  measure.count {i : α | ε ≤ a i} ≤ c / ε :=
+begin
+  rw ← lintegral_count at tsum_le_c,
+  apply (measure_theory.meas_ge_le_lintegral_div a_mble.ae_measurable ε_ne_zero ε_ne_top).trans,
+  exact ennreal.div_le_div tsum_le_c rfl.le,
+end
+
+/-- A sum of extended nonnegative reals which is finite can have only finitely many terms
+above any positive threshold.-/
+lemma finite_const_le_of_tsum_lt_top {a : α → ℝ≥0∞}
+  (tsum_ne_top : ∑' i, a i ≠ ∞) {ε : ℝ≥0∞} (ε_ne_zero : ε ≠ 0) :
+  {i : α | ε ≤ a i}.finite :=
+begin
+  have a_mble : @measurable _ _ ⊤ _ a, from measurable_space.top.measurable a,
+  by_cases ε = ∞,
+  { by_contra con,
+    rcases set.infinite.nonempty con with ⟨i, hi⟩,
+    have infty_le_ai : ∞ ≤ a i, by simpa only [h, mem_set_of_eq] using hi,
+    exact (lt_of_le_of_lt infty_le_ai (ennreal.lt_top_of_tsum_ne_top tsum_ne_top i)).ne rfl, },
+  apply (measure.count_apply_lt_top' measurable_space.measurable_set_top).mp,
+  have obs := @ennreal.count_const_le_le_of_tsum_le α ⊤ _ a a_mble _ rfl.le _ ε_ne_zero h,
+  exact lt_of_le_of_lt obs (ennreal.div_lt_top tsum_ne_top ε_ne_zero),
+end
+
+/-- Markov's inequality for `finset.card` and `tsum` in `ℝ≥0∞`. -/
+lemma finset_card_const_le_le_of_tsum_le {a : α → ℝ≥0∞}
+  {c : ℝ≥0∞} (c_ne_top : c ≠ ∞) (tsum_le_c : ∑' i, a i ≤ c)
+  {ε : ℝ≥0∞} (ε_ne_zero : ε ≠ 0) (ε_ne_top : ε ≠ ∞) :
+  ∃ hf : {i : α | ε ≤ a i}.finite, ↑hf.to_finset.card ≤ c / ε :=
+begin
+  by_cases ε = ∞,
```

Comment:
I think the assumption `ε_ne_top` is unnecessary considering your proof cases on `ε = ∞` anyway

Reply:
Thanks! Apparently I forgot to remove it once I included the handling of the `ε = ∞` case...